Build fails on HAIKU

### Checklist

- [x] I **have searched** the issue tracker and **was unable** to find an [open](../issues?q=is%3Aissue+is%3Aopen) or [closed](../issues?q=is%3Aissue+is%3Aclosed) issue matching what I'm seeing.
- [x] I **have verified** that the issue is happening in the **[latest nightly build](https://ci.86box.net/job/86Box/lastSuccessfulBuild/artifact/)**, or the latest commit, if building from source.

### What happened?

**Issue**
Building on Haiku (nightly, x86-64) fails in two places:

1. Compilation of src/qt/qt_platform.cpp fails in line 840 as pthread_setname_np is not declared (maybe a missing include)
2. Linking the final binary fails with a undefined reference to BMessage::FindInt32

I temporarily fixed the issues with:

1. Don't call pthread_setname_np
2. Add "-lbe" to the linker command

Otherwise 86Box is working perfectly on this platform!

**How to reproduce**

1. Install Haiku nightly x64
2. Install all the dependencies for building 86Box
3. Clone the master branch
4. Run the usual cmake commands to build

**Expected behavior**

Build works out of the box without manual intervention

### Configuration file

```ini
N/A, build issue
```

### Operating system

Haiku

### 86Box version

master branch f3a437c

### Build architecture

Windows - x64 (64-bit)

### Build type

- [ ] New recompiler
- [ ] Debug build

### Additional context

_No response_

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -108,6 +108,10 @@ endif()
 target_link_libraries(86Box cpu chipset mch dev mem fdd game cdrom zip mo hdd
     net print scsi sio snd vid voodoo plat ui)
 
+if(HAIKU)
+    target_link_libraries(86Box be)
+endif()
+
 if(WIN32 AND ARCH STREQUAL "i386")
     if(MINGW)
         target_link_options(86Box PRIVATE "LINKER:--large-address-aware")

--- a/src/qt/qt_platform.cpp
+++ b/src/qt/qt_platform.cpp
@@ -18,6 +18,11 @@
  *          Copyright 2021-2022 Cacodemon345
  *          Copyright 2021-2022 Teemu Korhonen
  */
+
+#ifdef __HAIKU__
+#include <OS.h>
+#endif
+
 #include <cstdio>
 
 #include <mutex>
@@ -828,6 +833,8 @@ plat_set_thread_name(void *thread, const char *name)
     pthread_setname_np(truncated);
 #    elif defined(Q_OS_NETBSD)
     pthread_setname_np(thread ? *((pthread_t *) thread) : pthread_self(), truncated, "%s");
+#    elif defined(__HAIKU__)
+    rename_thread(find_thread(NULL), truncated);
 #    elif defined(Q_OS_OPENBSD)
     pthread_set_name_np(thread ? *((pthread_t *) thread) : pthread_self(), truncated);
 #    else

--- a/src/unix/unix.c
+++ b/src/unix/unix.c
@@ -2,6 +2,9 @@
 #    define _FILE_OFFSET_BITS   64
 #    define _LARGEFILE64_SOURCE 1
 #endif
+#ifdef __HAIKU__
+#include <OS.h>
+#endif
 #include <SDL.h>
 #include <stdlib.h>
 #include <stdbool.h>
@@ -1398,16 +1401,20 @@ plat_set_thread_name(void *thread, const char *name)
     if (thread) /* Apple pthread can only set self's name */
         return;
     char truncated[64];
-#elif defined(Q_OS_NETBSD)
+#elif defined(__NetBSD__)
     char truncated[64];
+#elif defined(__HAIKU__)
+    char truncated[32];
 #else
     char truncated[16];
 #endif
     strncpy(truncated, name, sizeof(truncated) - 1);
 #ifdef __APPLE__
     pthread_setname_np(truncated);
-#elif defined(Q_OS_NETBSD)
+#elif defined(__NetBSD__)
     pthread_setname_np(thread ? *((pthread_t *) thread) : pthread_self(), truncated, "%s");
+#elif defined(__HAIKU__)
+    rename_thread(find_thread(NULL), truncated);
 #else
     pthread_setname_np(thread ? *((pthread_t *) thread) : pthread_self(), truncated);
 #endif

